test/integration/controllermanager/managedseedset/managedseedset_test.go

```diff
@@ -0,0 +1,368 @@
+				},
+			}
+			Expect(testClient.Status().Patch(ctx, &shootList.Items[0], patch)).To(Succeed())
+
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.List(ctx, managedSeedList, client.InNamespace(managedSeedSet.Namespace), client.MatchingLabelsSelector{Selector: selector})).To(Succeed())
+				g.Expect(managedSeedList.Items).To(HaveLen(1))
+			}).Should(Succeed())
+
+			By("Mark the ManagedSeed condition as SeedRegistered")
+			patch = client.MergeFrom(managedSeedList.Items[0].DeepCopy())
+			managedSeedList.Items[0].Status = seedmanagementv1alpha1.ManagedSeedStatus{
+				ObservedGeneration: managedSeedList.Items[0].GetGeneration(),
+				Conditions: []gardencorev1beta1.Condition{
+					{Type: seedmanagementv1alpha1.ManagedSeedSeedRegistered, Status: gardencorev1beta1.ConditionTrue},
+				},
+			}
+			Expect(testClient.Status().Patch(ctx, &managedSeedList.Items[0], patch)).To(Succeed())
+
+			By("Create Seed manually as ManagedSeed controller is not running in the test")
+			seed.Name = managedSeedList.Items[0].Name
+			Expect(testClient.Create(ctx, seed)).To(Succeed())
+			log.Info("Created Seed for ManagedSeed", "seed", client.ObjectKeyFromObject(seed), "managedSeed", client.ObjectKeyFromObject(&managedSeedList.Items[0]))
+
+			DeferCleanup(func() {
+				By("Delete Seed")
+				Expect(testClient.Delete(ctx, seed)).To(Or(Succeed(), BeNotFoundError()))
+				Eventually(func() error {
+					return testClient.Get(ctx, client.ObjectKeyFromObject(seed), seed)
+				}).Should(BeNotFoundError())
+			})
+
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.List(ctx, seedList, client.MatchingLabelsSelector{Selector: selector})).To(Succeed())
+				g.Expect(seedList.Items).To(HaveLen(1))
+			}).Should(Succeed())
+
+			By("Mark the Seed as Ready")
+			Expect(testClient.Get(ctx, client.ObjectKeyFromObject(seed), seed)).To(Succeed())
+			patch = client.MergeFrom(seed.DeepCopy())
+			seed.Status = gardencorev1beta1.SeedStatus{
+				ObservedGeneration: seed.GetGeneration(),
+				Conditions: []gardencorev1beta1.Condition{
+					{Type: gardencorev1beta1.SeedGardenletReady, Status: gardencorev1beta1.ConditionTrue},
+					{Type: gardencorev1beta1.SeedBootstrapped, Status: gardencorev1beta1.ConditionTrue},
+					{Type: gardencorev1beta1.SeedSystemComponentsHealthy, Status: gardencorev1beta1.ConditionTrue},
+					{Type: gardencorev1beta1.SeedBackupBucketsReady, Status: gardencorev1beta1.ConditionTrue},
+				},
+			}
+			Expect(testClient.Status().Patch(ctx, seed, patch)).To(Succeed())
+
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedSeedSet), managedSeedSet)).To(Succeed())
+				g.Expect(managedSeedSet.Status.PendingReplica).To(BeNil())
+				g.Expect(managedSeedSet.Status.ReadyReplicas).To(Equal(int32(1)))
+			}).Should(Succeed())
+		})
+	})
+
+	Context("scale-out", func() {
+		It("should create new replica and update the status.replicas field because spec.replicas value was increased", func() {
```

Comment:
When I tested this with the `stress` utility, this `It` kept flaking. Can you check?

Reply:
```
$ stress -ignore "unable to grab random port" -p 32 ./test/integration/controllermanager/managedseedset/managedseedset.test

22m50s: 1990 runs so far, 0 failures
22m55s: 2008 runs so far, 0 failures
23m0s: 2008 runs so far, 0 failures
```